worker/worker.go

```diff
@@ -45,13 +45,30 @@ func (w *Pool) StartWorkers() {
 				batchReadTime := time.Now()
 				//@TODO - Should add integration tests to prove that the worker receives the same message that it produced, on the delivery channel it created
 
+				eventCounts := make(map[string]int)
+				for _, e := range request.GetEvents() {
+					counts, ok := eventCounts[e.Type]
+					if !ok {
+						counts = 0
```

Comment:
Is this necessary. IINM, zero value applies to map. If so, you can increment it right away.

Reply:
Oh yeah, you are right.